### PR TITLE
agent: add flag to enable internal traffic policy

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

File: pkg/defaults/defaults.go [hunk 1]
```diff
@@ -595,6 +595,9 @@ const (
 
 	// NetNsPath is the default path to the mounted network namespaces directory
 	NetNsPath = "/var/run/cilium/netns"
+
+	// EnableIternalTrafficPolicy is the default value for option.EnableInternalTrafficPolicy
+	EnableInternalTrafficPolicy = true
```
Comment (joestringer) [inline pkg/defaults/defaults.go:603]: Should there be a corresponding change in `daemon/`?

Comment (nebril) [reply pkg/defaults/defaults.go:603]: There should! I should have marked this as draft, sorry.

**Review by asauber — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Review by ysksuzuki — APPROVED** [review]

**Comment by nebril** [comment]
rebased on main to resolve conflicts

**Pull request merged**